feat(curriculum): add string method workshop (#66936)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/blocks/workshop-employee-profile-validator/69e736c39dfa403496e8071e.md

---
id: 69e736c39dfa403496e8071e
title: Step 10
challengeType: 20
dashedName: step-10
---

# --description--

> The `.upper()` method converts all characters to `uppercase`. This is useful for name badges or headings:

Reviewer: ```suggestion
The `.upper()` method converts all characters to uppercase. This is useful for name badges or headings:
```

```py
name = 'sarah davis'
badge = name.upper()
print(badge)  # Output: SARAH DAVIS
```

Using `clean_name`, create a variable named `name_badge` by calling `.upper()` on it. 

> Then print the `name_badge` with the label `'Name badge: '`

Reviewer: ```suggestion
Using `clean_name`, create a variable named `name_badge` by calling `.upper()` on it.

Then print the `name_badge` with the label `'Name badge: '`.
```

---
id: 69e736c39dfa403496e8071e
title: Step 11
challengeType: 20
dashedName: step-11
---

# --description--

The `.upper()` method converts all characters to uppercase. This is useful for name badges or headings:

```py
name = 'sarah davis'
badge = name.upper()
print(badge)  # Output: SARAH DAVIS
```

> Using `clean_name`, create a variable named `name_badge` by calling `.upper()` on it.

Reviewer 2: ```suggestion
Create a variable named `name_badge` and assign it the result of calling the `.upper()` method on `clean_name`.
```

> Then print `name_badge` with the prefix `'Name badge: '`.

Reviewer 2: ```suggestion
Then print the result of concatenating the string `Name badge: ` and the `name_badge` variable.
```

> Then print the `name_badge` with the prefix `'Name badge: '`.

Reviewer 3: Matches "print `rejoined`" / "print `clean_email`" in steps 12, 15, and 16.

```suggestion
Then print `name_badge` with the prefix `'Name badge: '`.
```

# --hints--

You should create a variable named `name_badge`.

```js
({ test: () => assert(runPython(`_Node(_code).has_variable('name_badge')`)) })
```

> You should use `.upper()` on `clean_name` to assign `name_badge` a value.

Reviewer 2: ```suggestion
You should call `.upper()` on `clean_name` and assign the result to the `name_badge` variable.
```

```js
({ test: () => assert(runPython(`
node = _Node(_code).find_variable('name_badge')

> not node.is_empty() and node.is_equivalent("name_badge = clean_name.upper()")

Reviewer 2: ```suggestion
_Node(_code).find_variable('name_badge').is_equivalent("name_badge = clean_name.upper()")
```

`)) })
```

> You should print `name_badge` with the prefix `'Name badge: '`.

Reviewer 2: ```suggestion
You should print the result of concatenating the string `Name badge: ` and the `name_badge` variable.
```

---
id: 69e736c39dfa403496e8071e
title: Step 11
challengeType: 20
dashedName: step-11
---

# --description--

The `.upper()` method converts all characters to uppercase. This is useful for name badges or headings:

```py
name = 'sarah davis'
badge = name.upper()
print(badge)  # Output: SARAH DAVIS
```

Create a variable named `name_badge` and assign it the result of calling the `.upper()` method on `clean_name`.

> Then print `name_badge` so that the output reads: `Name badge: SARAH DAVIS`.

Reviewer 3: The trailing punctuation is the cross-step item: "so that the output reads" is followed by a colon and no period in steps 2, 3 and 5, a colon and a period in steps 6, 11, 13 and 16, and no colon with a period in steps 4, 9 and 15. Every hint line in the workshop uses the last of those, so aligning the descriptions to it is the smallest change.

```suggestion
Then print `name_badge` so that the output reads `Name badge: SARAH DAVIS`.
```
